frame/assets/src/lib.rs

```diff
@@ -114,214 +114,81 @@ mod benchmarking;
 pub mod weights;
 
 use sp_std::{fmt::Debug, prelude::*};
-use sp_runtime::{RuntimeDebug, traits::{
-	Member, AtLeast32BitUnsigned, Zero, StaticLookup, Saturating, CheckedSub, CheckedAdd
-}};
+use sp_runtime::{
+	RuntimeDebug,
+	traits::{
+		AtLeast32BitUnsigned, Zero, StaticLookup, Saturating, CheckedSub, CheckedAdd,
+	}
+};
 use codec::{Encode, Decode, HasCompact};
-use frame_support::{Parameter, decl_module, decl_event, decl_storage, decl_error, ensure,
-	traits::{Currency, ReservableCurrency, EnsureOrigin, Get, BalanceStatus::Reserved},
-	dispatch::{DispatchResult, DispatchError},
+use frame_support::{
+	ensure,
+	traits::{Currency, ReservableCurrency, BalanceStatus::Reserved},
+	dispatch::DispatchError,
 };
-use frame_system::ensure_signed;
 pub use weights::WeightInfo;
 
-type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub use pallet::*;
 
-/// The module configuration trait.
-pub trait Config: frame_system::Config {
-	/// The overarching event type.
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 
-	/// The units in which we record balances.
-	type Balance: Member + Parameter + AtLeast32BitUnsigned + Default + Copy;
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::DispatchResultWithPostInfo,
+		pallet_prelude::*,
+	};
+	use frame_system::pallet_prelude::*;
+	use super::*;
 
-	/// The arithmetic type of asset identifier.
-	type AssetId: Member + Parameter + Default + Copy + HasCompact;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
 
-	/// The currency mechanism.
-	type Currency: ReservableCurrency<Self::AccountId>;
+	#[pallet::config]
+	/// The module configuration trait.
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event> + IsType<<Self as frame_system::Config>::Event>;
```

Comment:
IsType already bounds Into
```suggestion
		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
```